torchrl/envs/transforms/transforms.py

```diff
@@ -2433,3 +2433,90 @@ def __repr__(self) -> str:
+    tensordict, this transform hos no effect.
+    """
+
+    inplace = True
+
+    def __init__(
+        self,
+        in_keys: Optional[Sequence[str]] = None,
+        out_keys: Optional[Sequence[str]] = None,
+    ):
+        """Initialises the transform. Filters out non-reward input keys and defines output keys."""
+        if in_keys is None:
+            in_keys = ["reward"]
+        else:
+            in_keys = [in_key for in_key in in_keys if in_key.startswith("reward")]
+
+        out_keys = [f"episode_{in_key}" for in_key in in_keys]
+
+        super().__init__(in_keys=in_keys, out_keys=out_keys)
+
+    def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Resets episode rewards."""
+        # Non-batched environments
+        if len(tensordict.batch_size) < 1 or tensordict.batch_size[0] == 1:
+            for out_key in self.out_keys:
+                if out_key in tensordict.keys():
+                    tensordict[out_key] = 0.0
+
+        # Batched environments
+        elif "reset_workers" in tensordict.keys():
+            for out_key in self.out_keys:
+                if out_key in tensordict.keys():
+                    tensordict[out_key][tensordict["reset_workers"]] = 0.0
+
+        return tensordict
+
+    def _call(self, tensordict: TensorDictBase) -> TensorDictBase:
+        """Updates the episode rewards with the step rewards."""
+        # Sanity checks
+        self._check_inplace()
+        for in_key in self.in_keys:
+            if in_key not in tensordict.keys():
+                return tensordict
+
+        # Update episode rewards
+        for in_key, out_key in zip(self.in_keys, self.out_keys):
+            reward = tensordict.get(in_key)
+            if out_key not in tensordict.keys():
+                tensordict.set(
+                    out_key, torch.zeros(*tensordict.shape, 1, dtype=reward.dtype)
+                )
+            tensordict[out_key] += reward
+
+        return tensordict
+
+    def transform_rward_spec(self, reward_spec: TensorSpec) -> TensorSpec:
+        """Transforms the reward spec such that the resulting spec matches transform mapping."""
+        # # Define episode_reward spec
+        # if isinstance(reward_spec, NdUnboundedContinuousTensorSpec):
+        #     episode_reward_spec = NdUnboundedContinuousTensorSpec(
+        #         shape=reward_spec.shape,
+        #         device=reward_spec.device,
+        #         dtype=reward_spec.dtype
+        #     )
+        # else:
+        #     episode_reward_spec = UnboundedContinuousTensorSpec(
+        #         device=reward_spec.device,
+        #         dtype=reward_spec.dtype
+        #     )
+        #
+        # # Add episode_reward spec to observation spec
+        # reward_spec = CompositeSpec(
+        #     reward=reward_spec,
+        #     episode_reward=episode_reward_spec
+        # )
+        return reward_spec
```

Comment:
I tried to modify the reward_spec, but that generated problem somewhere else in the code.

In any case, is episode_reward part of the reward?